tortoise/tortoise.go

```diff
@@ -525,7 +525,9 @@ func (t *turtle) computeEpochHeight(epoch types.EpochID) {
 	einfo := t.epoch(epoch)
 	heights := make([]uint64, 0, len(einfo.atxs))
 	for _, info := range einfo.atxs {
-		heights = append(heights, info.height)
+		if !info.malfeasant {
+			heights = append(heights, info.height)
+		}
 	}
 	einfo.height = getMedian(heights)
```

Comment:
say node A published its first atx for epoch X+1 in epoch X, then published its second atx for epoch X in the 2nd layer of epoch X+1. the epoch weight will not be deducted as a result of makrMalfeasant?

Reply:
i am confused by this 'timestamps', they have no impact whatsoever. 
if smesher submitted two atxs for the same epoch, first is counted towards the threshold, second is not.

i discovered when i actually need to revert ballots weights, and probably will revert all malfeseant atxs within the window as well, but this is not in this change

Reply:
the timing was based on how you calculated epoch weight. i recall the code calculated epoch weight at the first layer of every epoch.

maybe leave a TODO for reverting ballot/epoch weights?

Reply:
i will keep TODO on equivocation test, as it won't be reliable without reverting weight